src/utils/deconflictChunk.ts

```diff
@@ -227,10 +224,22 @@ function deconflictTopLevelVariables(
 					(variable instanceof ExportDefaultVariable && variable.getOriginalVariable() !== variable)
 				)
 			) {
+				const cachedSafeVariableName = Object.getOwnPropertyDescriptor(
+					module.info.safeVariableNames,
+					variable.name
+				)?.value;
+
+				if (cachedSafeVariableName && !usedNames.has(cachedSafeVariableName)) {
+					usedNames.add(cachedSafeVariableName);
+					variable.setRenderNames(null, cachedSafeVariableName);
+					continue;
+				}
```

Comment:
I see the check for `!usedNames.has(cachedSafeVariableName)` is the secret here to ensure everything is properly deconflicted. Nice work! So the effect is that you always avoid conflicts, but your variable names might "drift" over several builds. Which I think is fine as long as conflicts are always avoided.